### DISCUSSION TITLE
Issues with time series charts in Data explorer

### DISCUSSION THREAD

**Discussion by yanfeng1992** [apache/streampipes#2485] [Q&A] [answered]
I now have some mqtt data like this

![image](https://github.com/apache/streampipes/assets/89568107/76bd7ec8-631c-4202-aecd-498a17ce40d6)

I want to use time series data to display the randnum of each device in data explorer, using name_ as the field of group by
But I found that the name_ field cannot be displayed in the chart, and the randnum trend change of each device cannot be identified.
Is this a problem with my configuration, or is it a bug?

![image](https://github.com/apache/streampipes/assets/89568107/980d197c-fac6-4b5a-b9e0-a1aab67f0621)

![image](https://github.com/apache/streampipes/assets/89568107/caafd3ef-c4f2-4cbb-838a-82798b80a017)

**Comment by dominikriemer** [comment, marked as answer]
Hi @yanfeng1992 thanks for asking!
This could be a bug. The label should also show the group value.
We did a lot of work for the upcoming release to migrate all charts to Echarts. This should also fix your issue (see the screenshot below, colors for each group are slightly lightened and the correct label is shown). Also, grouping for other charts should work much better. If you want, you can give it a try and it would also great if you have any feedback what could be improved!

And welcome to the community :-)

![image](https://github.com/apache/streampipes/assets/33908476/57345126-463b-4627-b695-5d67bb69a6bf)

**Reply by yanfeng1992** [reply]
Thank you for your answer.  @dominikriemer 

I tried the new version which can solve the problem of group display

But I have a suggestion, can each grouped data use different colors. Now using the same colors, not easy to differentiate


![image](https://github.com/apache/streampipes/assets/89568107/5a64166d-3cf5-4c97-9a2d-f0708fc4117d)


By the way，when will version 0.95.0 be released?

**Comment by dominikriemer** [comment]
Thanks for trying @yanfeng1992!
I'm interested in your feedback concerning the colors:
Currently we chose to use a single color for all group members to make it clear when more than one field is selected. So each group member gets the same color with increased brightness (hard to see in the example). So would you prefer to adjust the brightness but still having the same color or to use different colors for each group member?

**Reply by yanfeng1992** [reply]
After your explanation, I prefer to adjust the brightness but still having the same color. But I hope the difference in brightness of each group can be more obvious. @dominikriemer
